common/dynamicconfig/constants.go

```diff
@@ -340,10 +340,16 @@ const (
 	// StandbyTaskMissingEventsDiscardDelay is the amount of time standby cluster's will wait (if events are missing)
 	// before discarding the task
 	StandbyTaskMissingEventsDiscardDelay = "history.standbyTaskMissingEventsDiscardDelay"
+	// QueuePendingTaskCriticalCount is the max number of pending task in one queue
+	// before triggering queue slice splitting and unloading
+	QueuePendingTaskCriticalCount = "history.queuePendingTaskCriticalCount"
 	// QueueReaderStuckCriticalAttempts is the max number of task loading attempts for a certain task range
 	// before that task range is split into a separate slice to unblock loading for later range.
 	// currently only work for scheduled queues and the task range is 1s.
 	QueueReaderStuckCriticalAttempts = "history.queueReaderStuckCriticalAttempts"
+	// QueuePendingTaskMaxCount is the max number of task pending tasks in one queue before stop
+	// loading new tasks into memory.
+	QueuePendingTaskMaxCount = "history.queuePendingTasksMaxCount"
```

Comment:
Add comment on what is the difference between this one and queuePendingTaskCriticalCount